ttnn/cpp/ttnn/graph_processor.hpp

```diff
@@ -0,0 +1,119 @@
+#pragma once
+
+#include "tt_metal/graph_tracking.hpp"
+#include "third_party/json/json.hpp"
+#include <mutex>
+#include <stack>
+#include <typeindex>
+#include <unordered_map>
+#include <functional>
+#include <any>
+
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+namespace ttnn {
+
+    class ProcessorHooks : public tt::tt_metal::IGraphHooks {
+    private:
+        bool do_block = false;
+    public:
```

Comment:
Nit on style. 
Can we agree in style to keep public first, private last with members first, methods last?